packages/core/src/request.ts

```diff
@@ -51,14 +51,18 @@ export function sessionToSentryRequest(session: Session, api: API): SentryReques
 
 /** Creates a SentryRequest from an event. */
 export function eventToSentryRequest(event: Event, api: API): SentryRequest {
-  // since JS has no Object.prototype.pop()
-  const { __sentry_samplingMethod: samplingMethod, __sentry_sampleRate: sampleRate, ...otherTags } = event.tags || {};
-  event.tags = otherTags;
-
   const sdkInfo = getSdkMetadataForEnvelopeHeader(api);
   const eventType = event.type || 'event';
   const useEnvelope = eventType === 'transaction';
 
+  const { transactionSampling, ...metadata } = event.debug_meta || {};
+  const { method: samplingMethod, rate: sampleRate } = transactionSampling || {};
+  if (Object.keys(metadata).length === 0) {
```

Comment:
Is `debug_meta` actually sent to Sentry? If not, maybe we can already drop it completely at this point?

Reply:
The new wasm integration uses it to send debug images: https://github.com/getsentry/sentry-javascript/blob/96e3ef3cae44fef06605330207e87173602d7742/packages/wasm/src/index.ts#L66-L67.

I don't know enough about wasm to know if you'd ever possibly have both, but I figured it was better to cover my bases.

Reply:
Ah, true. My bad.

Reply:
It's brand new, to be fair.